### DISCUSSION TITLE
Subgraph parallel execution

### DISCUSSION THREAD

**Discussion by steffyd** [dagster-io/dagster#21373] [Q&A] [answered]
I have a dynamic graph that executes a subgraph as follows:
```python
@graph
def subgraph_for_athlete_scoring(athlete):
    athlete_scores = ingest_athlete_and_score(athlete_to_score=athlete, mlflow_model=mlflow_model.to_source_asset())
    team_scores = fetch_team_scores_for_athlete(athlete_to_score=athlete)
    return combine_athlete_and_team_scores(athlete_scores, team_scores)

def athlete_team_scoring_graph(graph_name, description, config):
    @graph(name=graph_name, description=description, config=athlete_scoring_simplified_config)
    def athlete_team_scoring_graph():
        athletes_to_score = fetch_athletes_to_score()
        scores = (keys_to_score(athletes_to_score).map(
            lambda athlete: subgraph_for_athlete_scoring(athlete)
            )
        ).collect()
        return collect_athlete_metadata(scores)
    
    return athlete_team_scoring_graph.configured(name=graph_name, config_or_config_fn=config)
```
This graph is run in a job with the multiprocess executor as the executor definition. The behavior I'm looking for is for the subgraph to run in parallel. Essentially have each athlete's subgraph run simultaneously based on the max concurrency value. The behavior I'm getting instead is that all the ingest_athlete_and_score ops run in parallel, and then the fetch_team_scores_for_athlete ops. How can I enforce that the subgraph is run in parallel instead of all the same op for all athletes?

**Comment by sryza** [comment]
Hey @steffyd - have you seen https://github.com/dagster-io/dagster/discussions/8871?

**Reply by steffyd** [reply]
I have, however that didn't seem to work because I have two ops that need to run in parallel in the sub graph, then the downstream op needs to run in a single fanned out key. The tags would just keep the same behavior where the ops would still have all of a single op run, then the other op, for all keys.

**Reply by sryza** [reply, marked as answer]
Got it - in that case, I don't think this is currently possible in Dagster, so the best would be to file an issue with the request.
